src/HDF5.jl

```diff
@@ -957,6 +957,19 @@ function Base.read(obj::DatasetOrAttribute, ::Type{T}, I...) where T
     return val
 end
 
+"""
+    read!(obj::DatasetOrAttribute, output_buffer::AbstractArray{T} [, I...]) where T
+
+Read [part of] a dataset or attribute into a preallocated output buffer.
+The output buffer must be convertible to a pointer and have a contiguous layout.
+"""
+function Base.read!(obj::DatasetOrAttribute, buf::AbstractArray{T}, I...) where T
```

Comment:
I'm curious why do we need this method? Seems like `copyto!` alone would be sufficient as per the AbstractArray interface.

Reply:
We currently implement `Base.read`, so why not `Base.read!`?

Reply:
I don't think we should define this method, because we  are moving more and more having the `HDF5.Dataset` type  satisfying  the  `AbstractArray` interface. Users would expect  `copyto!`  as that is the  standard name for this function, thus additionally adding `Base.read!` seems unnecessary.  

I understand that we are defining methods on `Base.read` throughout the `HDF5` module, but I don't see that we should extend that and add `Base.read!` 

The definition in IO/Networking, also suggests that this wouldn't be the best overload, since it's used by default for reading from a filename or stream. 
https://docs.julialang.org/en/v1/base/io-network/#Base.read!

Reply:
`Base.read!` has been removed.